### PR TITLE
Calculate and set the SearchPreference cookie

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/search/search_service.dart [hunk 1]
```diff
@@ -550,15 +550,15 @@ class TagsPredicate {
   /// user-facing search query.
   Map<String, String> asSearchLinkParams() {
     final params = <String, String>{
-      'runtime': _tagPartsWithPrefix('runtime', value: true).join(' '),
-      'platform': _tagPartsWithPrefix('platform', value: true).join(' '),
+      'runtime': tagPartsWithPrefix('runtime', value: true).join(' '),
+      'platform': tagPartsWithPrefix('platform', value: true).join(' '),
```
Comment (jonasfj) [inline app/lib/search/search_service.dart]: Something missing here?

File: app/lib/frontend/handlers/listing.dart [hunk 2]
```diff
@@ -58,31 +60,31 @@ Future<shelf.Response> webPackagesHandlerHtml(shelf.Request request) async {
...
-  ));
+  final result = htmlResponse(
+    renderPkgIndexPage(
+      searchResult.packages,
+      links,
+      sdk: sdk,
+      searchQuery: searchQuery,
+      totalCount: totalCount,
+    ),
+    headers: createSearchPreferenceCookie(searchPreferences),
```
Comment (jonasfj) [inline app/lib/frontend/handlers/listing.dart]: Can this conflict with other cookies? Just thinking...

File: app/lib/search/search_service.dart (hunk 1 above)
Comment (isoos) [reply app/lib/search/search_service.dart]: Removed the leading `_` in the name, as I'm using outside of the class.

File: app/lib/frontend/handlers/listing.dart (hunk 2 above)
Comment (isoos) [reply app/lib/frontend/handlers/listing.dart]: We only set a single cookie at any response, and it does not causes conflicts with already set cookies.

Comment (jonasfj) [reply app/lib/frontend/handlers/listing.dart]: I see no reason to fear conflicts :)

File: app/lib/search/search_service.dart (hunk 1 above)
Comment (jonasfj) [reply app/lib/search/search_service.dart]: nvm

**Pull request merged**